### PR TITLE
Job timestamp should always be in milliseconds (fixed)

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

This PR always sets the job related timestamps in GCS to avoid timing inconsistencies and race conditions. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
cc @architkulkarni

**Comment by AmeerHajAli** [comment]
Can we make sure this is in 1.4.1?
Also, CC @ijrsvt

File: python/ray/tests/test_job.py [hunk 1]
```diff
@@ -118,10 +118,13 @@ def test_job_timestamps(ray_start_regular):
     jobs = list(ray.state.jobs())
     jobs.sort(key=lambda x: x["JobID"])
 
-    # jobs[0] is the test case driver.
+    driver = jobs[0]
     finished = jobs[1]
     running = jobs[2]
 
+    # The initial driver timestamp/start time go down a different code path.
+    assert driver["Timestamp"] == driver["StartTime"]
```
Comment (ijrsvt) [inline python/ray/tests/test_job.py:126]: Can we also do this for running!

**Review by ijrsvt — APPROVED** [review]
LGTM!

**Comment by wuisawesome** [comment]
Failing tests look flakey. Merging.

**Pull request merged**